### PR TITLE
Support 'generate_result' flag on parse(), search() and findall() methods

### PR DESCRIPTION
**Pull request by timofurrer**
If the ``generate_result`` flag is set to ``False`` a ``Match`` instance
is returned instead of a ``Result`` instance.

A ``Match`` instance has one method ``generate_result()`` which returns
a ``Result`` instance for this very match.

This feature is especially useful if you want to lazy evaluate the
custom type conversion function but check if it matches some string in
advanced.

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -144,8 +144,6 @@ Type  Characters Matched                          Output
       e.g. Sun Sep 16 01:03:52 1973
  th   HTTP log format date/time                   datetime
       e.g. 21/Nov/2011:00:07:11 +0000
- ts   Linux system log format date/time           datetime
-      e.g. Nov  9 03:37:44
```
Comment (r1chardj0n3s) [inline README.rst]: Why is this removed? This format is still supported.

**Comment by timofurrer** [comment]
Thanks for the review. I've renamed the method to `evaluate_result` and fixed the doctest builds which are currently failing on master.

**Comment by timofurrer** [comment]
Thanks! Would be nice to have a release for this :smiley: 

:beers:

**Comment by r1chardj0n3s** [comment]
Already up as 1.8.0 on https://pypi.python.org/pypi/parse :-D
